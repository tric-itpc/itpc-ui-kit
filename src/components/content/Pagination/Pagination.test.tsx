import React from "react"

import { fireEvent, render } from "@testing-library/react"

import { Pagination } from "./index"

describe("Компонент Pagination", () => {
  test("Снимок Pagination с данными", () => {
    const { asFragment } = render(
      <Pagination callback={() => {}} dataLength={100} step={10} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Рендер компонента Pagination", () => {
    const callbackMock = jest.fn()
    const { container } = render(
      <Pagination callback={callbackMock} dataLength={100} />
    )

    const pagination = container.querySelector(".itpc-pagination")
    expect(pagination).toBeInTheDocument()
  })

  test("Кнопка 'В начало' отключена, если на первой странице", () => {
    const callbackMock = jest.fn()
    const { container } = render(
      <Pagination callback={callbackMock} dataLength={100} />
    )

    const startButton = container.querySelector(".itpc-pagination__btn_left")
    expect(startButton).toBeDisabled()
  })

  test("Кнопка 'В конец' активна, если не на последней странице", () => {
    const callbackMock = jest.fn()
    const { container } = render(
      <Pagination callback={callbackMock} dataLength={100} />
    )

    const endButton = container.querySelectorAll(
      ".itpc-pagination__btn_right"
    )[1]
    expect(endButton).not.toBeDisabled()
  })

  test("Кнопка 'На следующую страницу' вызывает callback с правильными данными", () => {
    const callbackMock = jest.fn()
    const { container } = render(
      <Pagination callback={callbackMock} dataLength={100} step={10} />
    )

    const nextButton = container.querySelector(".itpc-pagination__btn_right")
    fireEvent.click(nextButton!)

    expect(callbackMock).toHaveBeenCalledWith({
      currentPage: 2,
      end: 20,
      start: 10,
    })
  })

  test("Кнопка 'На предыдущую страницу' вызывает callback с правильными данными", () => {
    const callbackMock = jest.fn()
    const { container } = render(
      <Pagination callback={callbackMock} dataLength={100} step={10} />
    )

    const nextButton = container.querySelector(".itpc-pagination__btn_right")
    fireEvent.click(nextButton!)

    const prevButton = container.querySelector(".itpc-pagination__btn_left")
    fireEvent.click(prevButton!)

    expect(callbackMock).toHaveBeenCalledWith({
      currentPage: 1,
      end: 10,
      start: 0,
    })
  })

  test("Кнопка перехода на конкретную страницу вызывает callback с правильными данными", () => {
    const callbackMock = jest.fn()
    const { getByText } = render(
      <Pagination callback={callbackMock} dataLength={100} step={10} />
    )

    const pageButton = getByText("3")
    fireEvent.click(pageButton)

    expect(callbackMock).toHaveBeenCalledWith({
      currentPage: 3,
      end: 30,
      start: 20,
    })
  })

  test("Правильное количество кнопок страниц отображается", () => {
    const callbackMock = jest.fn()
    const { container } = render(
      <Pagination callback={callbackMock} dataLength={50} step={10} />
    )

    const pageButtons = container.querySelectorAll(".itpc-pagination__btn_page")
    expect(pageButtons).toHaveLength(5)
  })
})
