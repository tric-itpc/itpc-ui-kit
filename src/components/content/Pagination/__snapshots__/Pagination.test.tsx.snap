// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Pagination Снимок Pagination с данными 1`] = `
<DocumentFragment>
  <div
    class="itpc-pagination"
  >
    <button
      class="itpc-pagination__btn itpc-pagination__btn_left"
      disabled=""
      type="button"
    >
      <i
        class="itpc-icon__double-arrow itpc-icon__double-arrow_orientation_left"
      >
        <svg
          height="8"
          viewBox="0 0 12 8"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
          />
        </svg>
        <svg
          height="8"
          viewBox="0 0 12 8"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
          />
        </svg>
      </i>
    </button>
    <button
      class="itpc-pagination__btn itpc-pagination__btn_left"
      disabled=""
      type="button"
    >
      <i
        class="itpc-icon__arrow itpc-icon__arrow_orientation_left"
      >
        <svg
          height="8"
          viewBox="0 0 12 8"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
          />
        </svg>
      </i>
    </button>
    <div
      class="itpc-pagination__pages"
    >
      <button
        class="itpc-pagination__btn itpc-pagination__btn_page itpc-pagination__btn_page_active"
        type="button"
      >
        1
      </button>
      <button
        class="itpc-pagination__btn itpc-pagination__btn_page"
        type="button"
      >
        2
      </button>
      <button
        class="itpc-pagination__btn itpc-pagination__btn_page"
        type="button"
      >
        3
      </button>
      <p
        class="itpc-pagination__text"
      >
        ...
      </p>
      <button
        class="itpc-pagination__btn itpc-pagination__btn_page"
        type="button"
      >
        10
      </button>
    </div>
    <button
      class="itpc-pagination__btn itpc-pagination__btn_right"
      type="button"
    >
      <i
        class="itpc-icon__arrow itpc-icon__arrow_orientation_right"
      >
        <svg
          height="8"
          viewBox="0 0 12 8"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
          />
        </svg>
      </i>
    </button>
    <button
      class="itpc-pagination__btn itpc-pagination__btn_right"
      type="button"
    >
      <i
        class="itpc-icon__double-arrow itpc-icon__double-arrow_orientation_right"
      >
        <svg
          height="8"
          viewBox="0 0 12 8"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
          />
        </svg>
        <svg
          height="8"
          viewBox="0 0 12 8"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
          />
        </svg>
      </i>
    </button>
  </div>
</DocumentFragment>
`;
